src/gpt/gpt_api.py

```diff
@@ -0,0 +1,46 @@
+import os
+import openai
+from openai.error import OpenAIError
+from src.config import Config
+
+
+try:
+    # Set OpenAI API key
+    api_key = Config().API_KEY
+    openai.api_key = api_key
+except:
+    print("OpenAI API key not found. Please set the environment variable OPENAI_API_KEY to your API key.")
+    exit(1)
+
+
+def request_chat_completion(previous_message:dict, role:str = "system",message :str = ""):
```

Comment:
Rather use enum for previous messages for making sure that the dict is of correct form